Handle null initial values properly in CollectionAttributeFlusher

When using `evm.convert` we set the initial value of attributes to null when using the MARK_NEW flag which we don't handle correctly in `CollectionAttributeFlusher` and maybe some other collection flushers as well. This should be fixed.

```
Caused by: java.lang.NullPointerException
	at com.blazebit.persistence.view.impl.update.flush.CollectionAttributeFlusher.getDirtyFlusher(CollectionAttributeFlusher.java:1210)
	at com.blazebit.persistence.view.impl.update.flush.CompositeAttributeFlusher.updateFlushers(CompositeAttributeFlusher.java:1214)
	at com.blazebit.persistence.view.impl.update.flush.CompositeAttributeFlusher.getNestedDirtyFlusher(CompositeAttributeFlusher.java:1194)
	at com.blazebit.persistence.view.impl.update.EntityViewUpdaterImpl.getNestedDirtyFlusher(EntityViewUpdaterImpl.java:607)
	at com.blazebit.persistence.view.impl.update.EntityViewUpdaterImpl.update(EntityViewUpdaterImpl.java:683)
	at com.blazebit.persistence.view.impl.update.EntityViewUpdaterImpl.executeUpdate(EntityViewUpdaterImpl.java:622)
	at com.blazebit.persistence.view.impl.EntityViewManagerImpl.update(EntityViewManagerImpl.java:1165)
	at com.blazebit.persistence.view.impl.EntityViewManagerImpl.update(EntityViewManagerImpl.java:1125)
	at com.blazebit.persistence.view.impl.EntityViewManagerImpl.save(EntityViewManagerImpl.java:1042)
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -17,6 +17,7 @@ Not yet released
 * Fix boot exception when using Hibernate `@Where` on entity level
 * Properly set `fromClassExplicitlySet` flag when using the subquery `from(String correlationPath)` methods
 * Fix concurrency issue during `EntityViewUpdaterImpl` initialization which can lead to a NPE
+* Fix NPE during flush for views that were converted with `ConvertOption.CREATE_NEW` and contain an updatable inverse mapping
 
 ### Backwards-incompatible changes
 

--- a/entity-view/impl/src/main/java/com/blazebit/persistence/view/impl/update/flush/CollectionAttributeFlusher.java
+++ b/entity-view/impl/src/main/java/com/blazebit/persistence/view/impl/update/flush/CollectionAttributeFlusher.java
@@ -1207,8 +1207,10 @@ public class CollectionAttributeFlusher<E, V extends Collection<?>> extends Abst
                         // TODO: should "replace" mean that the initial values we know right now are "removed" or the actual current values?
                         Map<Object, Object> added = Collections.emptyMap();
                         Map<Object, Object> removed = new IdentityHashMap<>();
-                        for (Object o : (Collection<?>) initial) {
-                            removed.put(o, o);
+                        if (initial != null) {
+                            for (Object o : (Collection<?>) initial) {
+                                removed.put(o, o);
+                            }
                         }
 
                         List<CollectionElementAttributeFlusher<E, V>> elementFlushers = getInverseElementFlushersForActions(context, (Collection<?>) current, added, removed);

